megamek/src/megamek/client/ui/swing/MovementDisplay.java

```diff
@@ -2934,6 +2936,24 @@ private synchronized void updateLoadButtons() {
         } // End ce-hasn't-moved
     } // private void updateLoadButtons
 
+    private void updateLayMineButton() {
+        final Entity ce = ce();
+        if (null == ce) {
+            return;
+        }
+
+        if (!ce.canLayMine()) {
+            setLayMineEnabled(false);
+        }
+        if (ce instanceof BattleArmor) {
+            setLayMineEnabled(cmd.getLastStep() == null
+                || cmd.isJumping()
+                || cmd.getLastStepMovementType().equals(EntityMovementType.MOVE_VTOL_WALK));
+        } else {
```

Comment:
This looks like, if the current entity can't lay a mine and isn't battle armor, the lay mine button will become enabled. Maybe you want a return after line 2946?

Reply:
Yup